api.js

```diff
@@ -14,73 +14,56 @@ function sendError(res, statusCode, code, description) {
 
 }
 
-app.get('/blobs/:blobId', (req, res) => {
-    db.query('select content_type, blob_value from blobs where blob_id = ?',
-        [req.params.blobId],
-        (err, data) => {
-            if (data.length != 1) {
-                sendError(res, 404, 'NotFound', 'Blob not found');
-            } else {
-                const row = data[0];
-                res.set('Content-Type', row.content_type);
-                res.send(row.blob_value);
-            }
-    });
+app.get('/blobs/:blobId', async (req, res) => {
+    const blobId = req.params.blobId;
+    const row = await db.oneOrNone(
+        'select content_type, blob_value from blobs where blob_id = $1',
+        [blobId]);

```

Comment:
En postgres se usa $1, $2, $3, ... en vez de ?, ?, ?, ...

Reply:
Ademas se usa await en vez de usar un callback (que en realidad es lo mismo, es lo que se llama syntactic sugar, te lo explico a mas detalle si quieres). La ventaja es que no acabas con la famosa "pyramid of doom" donde cada llamada a sql te agrega un callback, con su nesting, y se va haciendo la tal piramide. En vez, el resultado se regresa como resultado de la funcion async.